components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java

```diff
@@ -412,6 +412,24 @@ public Response read(String path) {
     }
 
     @POST
+    @Path("/workspace/read/sample")
```

Comment:
Is this to get all samples ? 

Then the method should be GET workspace/samples